addons/mail/static/src/utils/common/format.js

```diff
@@ -170,6 +171,65 @@ export function addLink(node, transformChildren) {
     return markup(node.outerHTML);
 }
 
+function formatThreadInMentionName(thread) {
+    const text = thread.parent_channel_id
+        ? `#${thread.parent_channel_id.displayName} > ${thread.displayName}`
+        : `#${thread.displayName}`;
+    return text;
+}
```

Comment:
We might want to just do a getter on thread instead.
Code in composer/navigable list should use the same getter.
`getMentionsFromText` also.

The `#` part should probably not be part of this function